pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -506,11 +506,8 @@ func initializeGlobalConfigHandles(ctx *baseOsMgrContext) {
 	if err != nil {
 		log.Fatal(err)
 	}
-	subGlobalConfig.MaxProcessTimeWarn = warningTime
-	subGlobalConfig.MaxProcessTimeError = errorTime
-	subGlobalConfig.ModifyHandler = handleGlobalConfigModify
-	subGlobalConfig.CreateHandler = handleGlobalConfigModify
-	subGlobalConfig.DeleteHandler = handleGlobalConfigDelete
+	subGlobalConfig.Handlers(handleGlobalConfigModify, handleGlobalConfigModify, handleGlobalConfigDelete, nil, nil)
+	subGlobalConfig.TimeLimits(warningTime, errorTime)
```

Comment:
Unrelated - but am wondering is we should have a better API for this - if you are touching this.. 

subParams = pubsub.SubParams{}
subParams.Handlers(handleGlobalConfigModify, handleGlobalConfigModify, handleGlobalConfigDelete, nil, nil)
subParams.TimeLimits(warningTime, errorTime)
subGlobalConfig, err := pubsub.Subscribe("", types.GlobalConfig{}, false, ctx, pubSubParams)

This allows all the parameters to be set when we actually create the Pubsub.. 
If we change them - we don't know when to make them active or when the Pubsub would notice that they have changed.. 

The above model says - When you are creating PubSub, specify all the params you want to set..

Reply:
Actually, I thought about that as well, some level of having a defined struct for it. The very minor problem is that you would need to modify all of the `Subscribe.*()` funcs; very minor.

My main motivation really was minimal changes in one PR. But the more I think about it, the more I realize that it is not that hard to do. OK, I like this one. I will adopt it.

Reply:
Done as you suggested.